UM/Settings/Models/SettingDefinitionsModel.py

```diff
@@ -291,26 +298,42 @@ def _getDefinitionsByKey(self, key: str) -> List["SettingDefinition"]:
 
         return self._container.findDefinitions(key = key)
 
+    def _expandRecursive(self, key: str) -> Set[str]:
+        definitions = self._getDefinitionsByKey(key)
+        if not definitions:
+            return set()
+
+        expanded_settings = {key}
+        for child in definitions[0].children:
+            expanded_settings.update(self._expandRecursive(child.key))
+
+        return expanded_settings
+
+
     @pyqtSlot(str)
-    def expandRecursive(self, key: str) -> None:
+    def expandRecursive(self, key: str, *, emit_signal: bool = True ) -> None:
         """Show the children of a specified SettingDefinition and all children of those settings as well."""
 
         definitions = self._getDefinitionsByKey(key)
         if not definitions:
             return
-        self.expand(key)
+
+        self._expanded.add(key)
 
         for child in definitions[0].children:
-            self.expandRecursive(child.key)
+            self.expandRecursive(child.key, emit_signal = False)
+
+        if emit_signal:
+            self.expandedChanged.emit()
+            self._scheduleUpdateVisibleRows()
 
     #@deprecated("Use collapseRecursive instead.", "4.5")  # Commented out because these two decorators don't work together.
     @pyqtSlot(str)
     def collapse(self, key: str) -> None:
         return self.collapseRecursive(key)
 
-    ##  Hide the children of a specified SettingDefinition and all children of those settings as well.
     @pyqtSlot(str)
-    def collapseRecursive(self, key: str) -> None:
+    def collapseRecursive(self, key: str, *, emit_signal: bool = True) -> None:
```

Comment:
A little bit of documentation is required as to why the boolean trap is created here.

Reply:
It's actually to prevent the boolean trap. I'll add the documentation.